python/python/res/simulator/robust_simulator.py

```diff
@@ -0,0 +1,121 @@
+from ecl.util import BoolVector
+
+from res.enkf import ErtRunContext, EnKFMain, EnkfConfigNode, EnkfNode, NodeId
+from res.server import SimulationContext
+from .robust_simulator_context import RobustContext
+
+class RobustSimulator(object):
+
+    def __init__(self, res_config, controls, results):
```

Comment:
Could we do some initial validation here of `controls` and `results`, ensuring that the first is indeed a dictionary mapping strings to lists of strings and the later a list of strings? It is at least unclear to me how this would go down if something else is provided and I think it is preferable to try to avoid the hard failures inside C when possible (especially for an API like this that is to be used mainly outside _libres_)..

Reply:
I have tried to test various incorrect input situations without actually resorting to `isinstance( )` - can of course extend that; but I do not see things actually entering the C layer - and crash hard - if the input is completely off. Might add more tests of course.

Reply:
Yes, you have very good both validation and testing for incorrect input to the `start`-method. But not for the initializer as far as I can see..

Reply:
> But not for the initializer as far as I can see..

No - that might be - will look into it.